src/System.Net.NameResolution/src/System/Net/DNS.cs

```diff
@@ -464,6 +473,8 @@ public static IPHostEntry EndGetHostByName(IAsyncResult asyncResult)
         {
             if (NetEventSource.Log.IsEnabled()) NetEventSource.Enter(NetEventSource.ComponentType.Socket, "DNS", "EndGetHostByName", asyncResult);
 
+            NameResolutionPal.EnsureSocketsAreInitialized();
```

Comment:
It shouldn't be required in the End* methods: it needs to be the first WinSock call (including the ones in System.Net.Sockets).